src/gpu/sycl/binary_kernels.hpp

```diff
@@ -146,6 +157,93 @@ struct binary_kernel_vec_t {
         return static_cast<float *>(src1_scale_.get_pointer());
     }
 
+    inline ::sycl::vec<float, 16> post_op_src_val(dim_t data_l_off) const {
+        ::sycl::vec<float, 16> post_po_sr;
+        const auto maxPostPo = conf_.post_ops.get_post_op();
+
+        for (dim_t po_idx = 0; po_idx < maxPostPo; po_idx++) {
+            float res = 0.0f;
+            if (po_idx == 0)
+                res = get_post_op_val(po1_src_, po_idx, data_l_off);
+            else if (po_idx == 1)
+                res = get_post_op_val(po2_src_, po_idx, data_l_off);
+            else if (po_idx == 2)
+                res = get_post_op_val(po3_src_, po_idx, data_l_off);
+            else if (po_idx == 3)
+                res = get_post_op_val(po4_src_, po_idx, data_l_off);
+            else if (po_idx == 4)
+                res = get_post_op_val(po5_src_, po_idx, data_l_off);
+
```

Comment:
```suggestion
            else
                assert(!"unexpected po index");
```

If asserts are supported in kernel.

Reply:
They are not.